ReactAndroid/src/main/java/com/facebook/react/views/scroll/ReactScrollView.java

```diff
@@ -341,4 +346,61 @@ protected void onOverScrolled(int scrollX, int scrollY, boolean clampedX, boolea
 
     super.onOverScrolled(scrollX, scrollY, clampedX, clampedY);
   }
+
+  @Override
+  public void onChildViewAdded(View parent, View child) {
+    setContentView(1 /* expectedChildCount */);
+  }
+
+  @Override
+  public void onChildViewRemoved(View parent, View child) {
+    setContentView(0 /* expectedChildCount */);
+
+  }
+
+  private void setContentView(int expectedChildCount) {
```

Comment:
nit: seems like we can just put part of this logic in onChildViewAdded and the rest in onChildViewRemoved. There are already assertions in android code that a scrollview can only have 1 child so I'm not really worried about that.